### DISCUSSION TITLE
Post json object

### DISCUSSION THREAD

**Discussion by czupet** [dotnet/yarp#333] [General] [answered]
On my destination Web Api application, I have the following method.

   [HttpPost]
        public ActionResult Post([FromBody] SomeRate rate)
        {
            return Ok($"Post works");
        }

When I call reverse-proxy I get 

{
    "type": "https://tools.ietf.org/html/rfc7231#section-6.5.13",
    "title": "Unsupported Media Type",
    "status": 415,
    "traceId": "|e685a43b-49e6605296249c50."
}

What am I doing wrong, or how do I call reverse-proxy with a JSON object?

if I have the following I don't get the error message.

   [HttpPost]
        public ActionResult Post()
        {
            return Ok($"Post works");
        }

This is my appsettings.json file

{
  "Logging": {
    "LogLevel": {
      "Default": "Information",
      "Microsoft": "Warning",
      "Microsoft.Hosting.Lifetime": "Information"
    }
  },
  "AllowedHosts": "*",
  "ReverseProxy": {
    "Routes": [
      {
        "RouteId": "app2",
        "ClusterId": "cluster1",
        "Match": {
          "Methods": [ "GET", "POST" ],
          "Hosts": [ "localhost" ],
          "Path": "/app2/"
        }
      },

      {
        "RouteId": "route2",
        "ClusterId": "cluster2",
        "Match": {
          "Hosts": [ "localhost" ]
        }
      }
    ],
    "Clusters": {
      "cluster1": {
        "LoadBalancing": {
          "Mode": "Random"
        },
        "Destinations": {
          "cluster1_destination1": {
            "Address": "https://localhost:44307/"
          },
          "cluster1_destination2": {
            "Address": "https://localhost:44307/"
          }
        }
      },
      "cluster2": {
        "Destinations": {
          "cluster2_destination1": {
            "Address": "localhost"
          }
        }
      }
    }
  }
}

Thank You
Peter

**Comment by Tratcher** [comment, marked as answer]
Are you using Kestrel or IIS? There's a known issue proxying POSTs through IIS (#255).

**Reply by czupet** [reply]
Yes, I am using IIS Express.
Thanks for the info.
